heracles_explained.md

```diff
@@ -1,22 +1,92 @@
 ## Analysis of Heracles.ts 
 
+HydraClient, also known as Heracles.ts, is a generic client for Hydra-powered Web APIs.
+
 ### Client Part in Hydra 
 
+The basic idea behind Hydra is to provide a vocabulary which enables a server to advertise valid state transitions to a client. A client can then use this information to construct HTTP requests which modify the server’s state so that a certain desired goal is achieved. Since all the information about the valid state transitions is exchanged in a machine-processable way at runtime instead of being hardcoded into the client at design time, clients can be decoupled from the server and adapt to changes more easily.
 
+Heracles.ts defines mainly two type of Index:
 
+* Classes
+* Interface
 
+and it also contain a vaiable called `hydra` like `hydra:{any_string("http://www.w3.org/ns/hydra/core#")}`.
 
 
 
 ### Components Of Client 
 
+Classes are further divides in:
+* HydraClient
+* ApiDocumentation
+* JsonLdHypermediaProcessor
+* ResourceEnrichmentProvider
+
+Interface are divided in:
+* IClass
+* IApiDocumentation
+* IHydraResource
+* IHypermedia
+* IHypermediaContainer
+* IHypermediaProcessor
+* IOperation
+* IResource
+* IWebResource
+
```

Comment:
@sandeepsajan0 most part of it is copied from Heracles docs we can just look it there.
You should focus more on things like how the client works, what each of the above-specified classes and Interfaces does, how can we use this to improve our agent etc.

Also, the English you write should be grammatically correct. You can use [Grammerly](https://www.grammarly.com) for this if you want to.

Reply:
okay.